### PR TITLE
Switch the bitcoin payment system to match the rpcwallet interface.

### PR DESCRIPTION
**Pull request by c00w**
This is mostly a refactor so that we're no longer sending bitcoin payments over the wire thus allowing us to use bitcoins with less insane amounts of work required.

### CODE REVIEW

File: ledger.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+		p.l.Lock()
+		p.packets[d.hash] = d
+		p.l.Unlock()
+	}
+}
+
+func (p *ledger) RecordPayment(y Payment) {
+	p.l.Lock()
+	defer p.l.Unlock()
+	p.incoming_debt[y.Source] = payDebt(p.incoming_debt[y.Source], y.Amount)
```
Comment (eoswald) [inline ledger.go:122]: Why do you pay the debt owed to you?

File: node.go [hunk 2]
```diff
@@ -71,16 +71,28 @@ func (n *node) sendPayments() {
...
+				n.Router.RecordPayment(Payment{n.id.PublicKey().Hash(), c, owed})
+				n.Router.SendPaymentHash(c, p)
 			}
 		}
 		n.l.Unlock()
 	}
 }
 
+func (n *node) receivePayments() {
+	for h := range n.Router.PaymentHashes() {
```
Comment (eoswald) [inline node.go:86]: Where is PaymentHashes() implemented?

File: ledger.go (hunk 1 above)
Comment (c00w) [inline ledger.go:122]: This handles both directions of payments, and I think what I did was be slightly clever and handle both.

File: node.go (hunk 2 above)
Comment (c00w) [inline node.go:86]: It's implemented in payment.go, specificallly the payment handler inside the router.

Comment (eoswald) [inline node.go:86]: Sorry didn't see that, searching the repo only searches master.

**Pull request merged**